src/optimizer/pullup/pullup_filter.cpp

```diff
@@ -0,0 +1,27 @@
+#include "duckdb/optimizer/filter_pullup.hpp"
+#include "duckdb/planner/operator/logical_filter.hpp"
+#include "duckdb/planner/expression/bound_comparison_expression.hpp"
+#include "duckdb/planner/expression_iterator.hpp"
+#include "duckdb/planner/expression/bound_between_expression.hpp"
+
+namespace duckdb {
+using namespace std;
+
+unique_ptr<LogicalOperator> FilterPullup::PullupFilter(unique_ptr<LogicalOperator> op) {
+	D_ASSERT(op->type == LogicalOperatorType::LOGICAL_FILTER);
+
+	if(can_pullup) {
+		unique_ptr<LogicalOperator> child = move(op->children[0]);
+		child = Rewrite(move(child));
+		//moving filter's expressions
+		for(idx_t i=0; i < op->expressions.size(); ++i) {
+			filters_expr_pullup.push_back(move(op->expressions[i]));
+		}
+		op.reset(nullptr); //deleting logical filter
```

Comment:
This is not necessary; `op` will be automatically deleted by going out of scope.